src/core/pathname.cc

```diff
@@ -988,11 +989,12 @@ CL_DEFUN Pathname_sp cl__pathname(T_sp x) {
     x = cl__parse_namestring(x);
   } else if (gc::IsA<Pathname_sp>(x)) {
     // do nothing
-  } else if (gc::IsA<Stream_sp>(x)) {
+  } else if ((gc::IsA<FileStream_sp>(x)) or (gc::IsA<SynonymStream_sp>(x))){
```

Comment:
I think this will be the first 'or' in the Clasp code. 
Should we allow this?  Or stick with || ???